rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+---
+title: 'Nescience - A user-centric state separation architecture'
+date: 2024-08-07 12:00:00
+authors: moudy
+published: false
+slug: Nescience-A-state-separation-architecture
+categories: research
+
+
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+---
+
+Nescience, a user-centric state separation architecture.
+
+<!--truncate-->
+
+# A. Introduction
+
+In blockchain applications, privacy settings are typically predefined by developers, leaving users with limited control. This traditional, 
+one-size-fits-all approach often leads to inefficiencies and potential privacy concerns, as it fails to cater to the diverse needs of individual users. 
+The Nescience State Separation Architecture (NSSA) aims to address these issues by shifting privacy control from developers to users. NSSA introduces a flexible, 
+user-centric approach that allows for customized privacy settings to better meet individual needs. This blog post will delve into the details of NSSA, 
+including its different execution types, cryptographic foundations, and unique challenges.
+
+### Introducing NSSA: A User-Centric Approach
+
+NSSA gives users control over their privacy settings by introducing shielded (which create a layer of privacy for the outputs and only the necessary details are shared) 
+and deshielded (which reveal private details making them publicly visible) execution types, in addition to the traditional public and private modes. This flexibility allows 
+users to customize their privacy settings to match their unique needs, whether they require high levels of confidentiality or more transparency. In NSSA, the system is divided 
+into two states: public and private. The public state uses an account-based model, while the private state employs a UTXO-based model. Private executions within NSSA utilize 
+UTXO exchanges, ensuring that transaction details remain confidential. The sequencer verifies these exchanges without accessing specific details, enhancing privacy by unlinking 
+sender and receiver identities. Zero-Knowledge Proofs (ZKPs) allow users to prove transaction validity without revealing data, maintaining the integrity and confidentiality of 
+private transactions. UTXOs contain assets such as balances, NFTs, or private storage data, and are stored in plaintext within Sparse Merkle Trees (SMTs) in the private state and 
```

Comment:
Replace: 
"Sparse Merkle Trees" 

With: 
"sparse Merkle tree" 

(it's not a proper noun - apparently https://eprint.iacr.org/2016/683.pdf?ref=hackernoon.com )

Reply:
In the paper you mentioned they refer to a single sparse merkle tree while in the blogpost we refer to several.

Reply:
apologies, the point was the capitalisation, I accidentally omitted the "s", it should be "sparse Merkle trees"